### PR TITLE
Add SERVER prefix to environment variable names

### PR DESCRIPTION
**Pull request by fmoor**
fixes https://github.com/edgedb/edgedb/issues/2643

This change disambiguates server environment variables from client environment variables.

### CODE REVIEW

**Comment by fmoor** [comment]
I didn't add `SERVER` to the database specific options. Is this what you had in mind @elprans?

**Comment by elprans** [comment]
> I didn't add `SERVER` to the database specific options

All options should be prefixed.  Please also add fallback to old vars and issue a deprecation warning in `edbdocker_lookup_env_var`

File: docker-entrypoint-funcs.sh [hunk 1]
```diff
@@ -135,87 +135,133 @@ edbdocker_run_entrypoint_parts() {
...
+    edbdocker_log "${msg[@]}"
+  fi
+
+  local -A exclusive_pairs=(
+    ["${var}"]="${file_var}"
+    ["${var}"]="${old_var}"
+    ["${var}"]="${old_file_var}"
+    ["${old_var}"]="${file_var}"
+    ["${old_var}"]="${old_file_var}"
+    ["${file_var}"]="${old_file_var}"
```
Comment (fmoor) [inline docker-entrypoint-funcs.sh:247]: This implementation will not check for usage of mutually exclusive variable pairs in any particular order. Is that acceptable?

**Review by elprans — APPROVED** [review]

**Comment by elprans** [comment]
I've made some fixups to the patch and undeprecated the unprefixed HOST|PORT|PASSWORD variables, since those aren't ambiguous and make it easier to share values in compose.

**Pull request merged**